api/handlers.go

```diff
@@ -87,11 +88,56 @@ func (api *API) loginPOST(w http.ResponseWriter, req *http.Request, _ httprouter
 func (api *API) loginPOSTCredentials(w http.ResponseWriter, req *http.Request, email, password string) {
 	// Fetch the user with that email, if they exist.
 	u, err := api.staticDB.UserByEmail(req.Context(), email, false)
-	if err != nil {
+	if err != nil && !errors.Contains(err, database.ErrUserNotFound) {
 		api.staticLogger.Tracef("Error fetching a user with email '%s': %+v\n", email, err)
 		api.WriteError(w, err, http.StatusUnauthorized)
 		return
 	}
+	// If the user is not in the DB there is a chance that it's in CockroachDB.
+	// Try to fetch it from there and if it's there, create a version of it in
+	// MongoDB. Then try to login against this new user record, as the hash of
+	// the password will be moved from CockroachDB and fully usable.
+	if errors.Contains(err, database.ErrUserNotFound) {
+		cru, err := database.CockroachUserByEmail(api.staticCockroachDB, email)
+		if err != nil {
+			api.staticLogger.Warnf("Failed to fetch user from CockroachDB: %s", err)
+			api.WriteError(w, database.ErrUserNotFound, http.StatusUnauthorized)
+			return
+		}
+		u = &database.User{
+			Email:        cru.Email,
+			PasswordHash: cru.PassHash,
+			Sub:          cru.Sub,
+			CreatedAt:    cru.CreatedAt,
+		}
+		// We need to generate an ID for the user because UserSave won't be able
+		// to create one for us.
+		copy(u.ID[:], fastrand.Bytes(len(u.ID)))
+		err = api.staticDB.UserSave(req.Context(), u)
+		if err != nil {
+			api.staticLogger.Warnf("Failed to move user from CockroachDB: %s", err)
+			api.WriteError(w, database.ErrUserNotFound, http.StatusUnauthorized)
+			return
+		}
+		api.staticLogger.Debugf("User with email '%s' moved from CockroachDB.", u.Email)
+	}
+	// Check if we somehow have an empty password hash and try to fill that.
```

Comment:
What is the edge case that would cause this? 

Looking at the code it shouldn't be because it was blank in cockroach, as you just pulled it so pulling it again would also be blank.

If it is blank from mongo, why wouldn't that just be an error?